### PR TITLE
feat: update deprecated type change

### PR DESCRIPTION
**Pull request by tobySolutions**
Worked with @chibuike-19 to update `never` types in the type ternaries across the core table code with Jsdoc's `@deprecated` block tags by creating interfaces for the  unavailable options.

- [x] TableState
- [ ] TableOptions
- [ ] ColumnDefBase

### CODE REVIEW

File: packages/table-core/src/features/column-filtering/ColumnFiltering.types.ts [hunk 1]
```diff
@@ -222,6 +222,16 @@ export interface TableOptions_ColumnFiltering<
 > extends ColumnFiltersOptionsBase<TFeatures, TData>,
     ResolvedFilterFns<TFeatures, TData> {}
 
+export interface TableOptions_ColumnFiltering_Unavailable<
```
Comment (KevinVandy) [inline packages/table-core/src/features/column-filtering/ColumnFiltering.types.ts:225]: I think we're going to have to make much more in-depth changes for table options. This PR can focus just on the state type changes.

Comment (KevinVandy) [reply packages/table-core/src/features/column-filtering/ColumnFiltering.types.ts:225]: I'm checking out the branch and removing this part. The needed change in a future pr will be to mark each of the `ColumnFiltersOptionsBase` interface properties as deprecated with jsdoc, though still with their type, when their associated feature is not present

Comment (tobySolutions) [reply packages/table-core/src/features/column-filtering/ColumnFiltering.types.ts:225]: Sorry, how do you mean @KevinVandy? Should we do away with all work done on `TableOptions`? I wasn't really sure of our work there though.

Is it okay to merge now with all that work stripped out?

**Review by KevinVandy — APPROVED** [review]

File: packages/table-core/src/features/column-filtering/ColumnFiltering.types.ts (hunk 1 above)
Comment (KevinVandy) [reply packages/table-core/src/features/column-filtering/ColumnFiltering.types.ts]: I just reverted those parts and added the commit to this pr

Comment (tobySolutions) [reply packages/table-core/src/features/column-filtering/ColumnFiltering.types.ts]: Awesome! Thanks Kevin!!

**Pull request merged**